cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -527,27 +528,42 @@ Future<Void> rollingUpdateForNewCaKey() {
         }
     }
 
-    /* test */ Future<Set<NodeRef>> getKafkaReplicas() {
+    /* test */ Future<List<StrimziPodSet>> updateKafkaPodCertAnnotations() {
         Labels selectorLabels = Labels.EMPTY
                 .withStrimziKind(reconciliation.kind())
                 .withStrimziCluster(reconciliation.name())
                 .withStrimziName(KafkaResources.kafkaComponentName(reconciliation.name()));
-
         return strimziPodSetOperator.listAsync(reconciliation.namespace(), selectorLabels)
                 .compose(podSets -> {
-                    Set<NodeRef> nodes = new LinkedHashSet<>();
-
                     if (podSets != null) {
-                        for (StrimziPodSet podSet : podSets) {
-                            nodes.addAll(ReconcilerUtils.nodesFromPodSet(podSet));
-                        }
+                        List<StrimziPodSet> updatedPodSets = podSets
+                                .stream()
+                                .map(podSet -> WorkloadUtils.patchAnnotations(
+                                        podSet,
+                                        Map.of(
+                                                Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, String.valueOf(clusterCa.caCertGeneration()),
+                                                Ca.ANNO_STRIMZI_IO_CLUSTER_CA_KEY_GENERATION, String.valueOf(clusterCa.caKeyGeneration()),
+                                                Ca.ANNO_STRIMZI_IO_CLIENTS_CA_CERT_GENERATION, String.valueOf(clientsCa.caCertGeneration())
+                                        )))
+                                .toList();
+                        return strimziPodSetOperator.batchReconcile(
+                                reconciliation,
+                                reconciliation.namespace(),
+                                updatedPodSets,
+                                selectorLabels
+                        ).map(i -> updatedPodSets);
```

Comment:
+1 ... Kate, out of curiosity, what was the reasoning to return the StrimziPodSet instead?

Reply:
I went back and forth a few times about how to structure this. Part of the motivation was that returning the StrimziPodSet makes it easier to test with the existing tests. But also it felt a bit strange to return a `Set<NodRef>` from a method that is updating annotations. I would only be returning the `Set<NodeRef>` because I know what method is going to be called next, rather than it making sense for this method on it's own. The StrimziPodSet has been patched, even though the nodes haven't been rolled yet. I can change it to Set<NodeRef> and update the tests accordingly if you prefer

Reply:
I don't know. I think having the encapsulation and not leaking the out-of-date StrimziPodSet would be more important for me.

> But also it felt a bit strange to return a Set<NodRef> from a method that is updating annotations.

I guess this can be solve with a different name? `patchCaGenerationAndReturnNodes`?

Reply:
I've updated it to use the name you suggested and altered the tests